build/build-image/cross/Dockerfile

```diff
@@ -15,15 +15,17 @@
 # This file creates a standard build environment for building cross
 # platform go binary for the architecture kubernetes cares about.
 
-FROM  golang:1.4.2
-MAINTAINER  Joe Beda <jbeda@google.com>
+FROM  golang:1.5.3
+# Original MAINTAINER  Joe Beda <jbeda@google.com>
+MAINTAINER Brendan Burns <bburns@google.com>
 
 ENV KUBE_CROSSPLATFORMS \
   linux/386 linux/arm \
   darwin/amd64 darwin/386 \
   windows/amd64 windows/386
 
-RUN cd /usr/src/go/src && for platform in ${KUBE_CROSSPLATFORMS}; do GOOS=${platform%/*} GOARCH=${platform##*/} ./make.bash --no-clean; done
+# Pre-compile the standard go library when cross-compiling. This is much easier now when we have go1.5
+RUN for platform in ${KUBE_CROSSPLATFORMS}; do GOOS=${platform%/*} GOARCH=${platform##*/} go install std; done
```

Comment:
The complaint on this site looks to be an issue only for homedir setups (vs a container):

> This logic also holds true for the standard library, which lives in /usr/local/go/src, so will be compiled to /usr/local/go/pkg/$GOOS_$GOARCH. This is a problem, because when cross compiling the go tool needs to rebuild the standard library for your target, but the binary distribution expects that /usr/local/go is not writeable.